File: typescript/math/pascal.ts
// Print 10 rows of Pascal's triangle, each row summed from the previous one shifted both ways.
// Run: tsc --target es2020 pascal.ts && node pascal.js

const rows = 10;

let row: number[] = [1];

for (let i = 0; i < rows; i++) {
    console.log(row.join(" "));
    const shifted = [0, ...row];
    const padded = [...row, 0];
    row = shifted.map((value, index) => value + padded[index]);
}

File: dart/math/pascal.dart
// Print 10 rows of Pascal's triangle, each row summed from the previous one shifted both ways.
// Run: dart run pascal.dart

const rows = 10;

void main() {
  var row = <int>[1];

  for (var i = 0; i < rows; i++) {
    print(row.join(' '));

    final shifted = [0, ...row];
    final padded = [...row, 0];
    row = [for (var j = 0; j < shifted.length; j++) shifted[j] + padded[j]];
  }
}
